helpers/elixir/bin/check_update.exs

```diff
@@ -19,7 +19,7 @@ defmodule UpdateChecker do
 
       {:ok, {:error, error}} -> {:error, error}
 
-      nil -> {:error, :dependency_resolution_timed_out}
+      {:ok, nil} -> {:error, :dependency_resolution_timed_out}
```

Comment:
Proper fix is actually on line 7, I shouldn't have removed the chained `Task.shutdown(task)`

```elixir
task = Task.async(fn -> do_resolution(dependency_name) end)
case Task.yield(task, 30000) || Task.shutdown(task) do
  {:ok, {:ok, :resolution_successful}} ->
    ...
  {:ok, {:error, error}} -> {:error, error}

  nil -> {:error, :dependency_resolution_timed_out}

  {:exit, reason} -> {:error, reason}
end
```

Reply:
Awesome, thanks!